docs/reverse_proxy.md

```diff
@@ -42,23 +42,53 @@ the reverse proxy and the homeserver.
             location /_matrix {
                 proxy_pass http://localhost:8008;
                 proxy_set_header X-Forwarded-For $remote_addr;
+            }            
+            location /.well-known/matrix/server {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
+            }
+            location /.well-known/matrix/client {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
+            }
+            location /.well-known {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
             }
         }
 
         server {
-            listen 8448 ssl default_server;
-            listen [::]:8448 ssl default_server;
+            listen 8448 ssl;
+            listen [::]:8448 ssl;
             server_name example.com;
 
             location / {
                 proxy_pass http://localhost:8008;
                 proxy_set_header X-Forwarded-For $remote_addr;
+            }            
+            location /.well-known/matrix/server {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
+            }
+            location /.well-known/matrix/client {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
+            }
+            location /.well-known {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
+            }
+            location /_matrix {
+                proxy_pass http://localhost:8008;
+                proxy_set_header X-Forwarded-For $remote_addr;
             }
```

Comment:
these are redundant to the `location /` definition.

In any case .well-known is never used on port 8448.

Reply:
You can't expect `location /` to be occupied exclusively by synapse. Many do point it their other federated social instances for example.

Reply:
firstly, I *can* expect it to be used exclusively by synapse, on port 8448.

but even if what you say is true, you haven't removed the `location /` definition. There's no point in adding these new paths while the old one is still there.